tests/integration/services/test_k8s.py

```diff
@@ -0,0 +1,401 @@
+            name=given_name,
+        )
+
+        # then
+        # the return type should be a dict.
+        assert isinstance(result, dict)
+
+        assert result["kind"] == given_kind
+        assert result["apiVersion"] == given_api_version
+        assert result["metadata"]["name"] == given_name
+        assert result["metadata"]["namespace"] == given_namespace
+        if result["kind"] == "Secret":
+            assert result["data"] == {}
+
+    @pytest.mark.parametrize(
+        "given_api_version, given_kind, given_namespace, given_name",
+        [
+            # Test case: should be able to describe a pod.
+            (
+                "apps/v1",
+                "Deployment",
+                "nginx-wrong-image",
+                "nginx",
+            ),
+            # Test case: should be able to describe Secret.
+            (
+                "v1",
+                "Secret",
+                "kyma-system",
+                "eventing-nats-secret",
+            ),
+            # Test case: should be able to describe Eventing CR.
+            (
+                "operator.kyma-project.io/v1alpha1",
+                "Eventing",
+                "kyma-system",
+                "eventing",
+            ),
+            # Test case: should be able to describe NATS CR.
+            (
+                "operator.kyma-project.io/v1alpha1",
+                "NATS",
+                "kyma-system",
+                "eventing-nats",
+            ),
+            # Test case: should be able to describe Function CR.
+            (
+                "serverless.kyma-project.io/v1alpha2",
+                "Function",
+                "kyma-app-serverless-syntax-err",
+                "func1",
+            ),
+        ],
+    )
+    def test_describe_resource(
+        self, k8s_client, given_api_version, given_kind, given_namespace, given_name
+    ):
+        # when
+        result = k8s_client.describe_resource(
+            api_version=given_api_version,
+            kind=given_kind,
+            namespace=given_namespace,
+            name=given_name,
+        )
+
+        # then
+        # the return type should be a dict.
+        assert isinstance(result, dict)
+
+        assert result["kind"] == given_kind
+        assert result["apiVersion"] == given_api_version
+        assert result["metadata"]["name"] == given_name
+        assert result["metadata"]["namespace"] == given_namespace
+        if result["kind"] == "Secret":
+            assert result["data"] == {}
+
+        assert isinstance(result["events"], list)
+        if len(result["events"]) > 0:
+            for item in result["events"]:
+                assert isinstance(item, dict)
+
+    @pytest.mark.parametrize(
+        "given_namespace",
+        [
+            # Test case: should be able to get list of not running pods.
+            "nginx-wrong-image",
+            # Test case: should be able to get list of not running pods in all namespaces.
+            "",
```

Comment:
should we verify in the test logic that it is really retrieving all the pods in all namespace? With pods list size, for example.

Reply:
Well, I started with that approach with a k3d cluster but realized that its duplicate work which we have already done in evaluation tests. We only need to read data, if we would have needed to update resources then having a separate isolated k8s env would be helpful. We can for now keep this and do improvements in a separate issue after discussing with the team.

Reply:
I see, we have to make sure the test data is not changed that may break the tests.

Reply:
I will add documentation in a separate PR.